modules/v2x/v2x_proxy/app/v2x_proxy.cc

```diff
@@ -13,147 +13,201 @@
+    return false;
   }
-  first_flag_reader_ = node_->CreateReader<StatusResponse>(
-      "/apollo/v2x/inner/sync_flag",
-      [this](const std::shared_ptr<const StatusResponse>& msg) {
-        x2v_trafficlight_timer_->Start();
-      });
-  if (first_flag_reader_ == nullptr) {
-    AERROR << "Create sync flag reader failed";
-    exit(1);
+  bool flag = false;
+  std::string str_one;
+  while (getline(input_file, str_one)) {
+    flag = true;
+    whitelist->insert(str_one);
   }
-  // x2v_trafficlight_timer_->Start();
-  v2x_carstatus_timer_->Start();
-
-  init_flag_ = true;
+  return flag;
 }
 
 bool V2xProxy::InitFlag() { return init_flag_; }
 
-bool V2xProxy::TrafficLightProc(CurrentLaneTrafficLight* msg) {
-  if (!msg->has_gps_x_m() || !msg->has_gps_y_m()) {
-    AERROR << "Error::v2x trafficlight ignore, gps point is null";
-    return false;
-  }
-  apollo::common::PointENU point;
-  point.set_x(msg->gps_x_m());
-  point.set_y(msg->gps_y_m());
-  std::vector<apollo::hdmap::SignalInfoConstPtr> signals;
-  if (hdmap_->GetForwardNearestSignalsOnLane(point, 1000.0, &signals) != 0) {
-    AERROR << "Error::v2x trafficlight ignore, hdmap get no signals";
-    AERROR << "traffic light size : " << signals.size();
-    return false;
-  }
-  for (auto i = signals.begin(); i != signals.end(); i++) {
-    if ((*i)->id().id().empty()) {
-      AERROR << "Error::v2x trafficlight ignore, signals id is empty";
-      return false;
+void V2xProxy::RecvOsPlanning() {
+  auto planning_adc = std::make_shared<::apollo::planning::ADCTrajectory>();
+  auto res_light =
+      std::make_shared<::apollo::perception::TrafficLightDetection>();
+  os_interface_->GetPlanningAdcFromOs(planning_adc);
+  // OK get planning message
+  std::shared_ptr<OSLight> last_os_light = nullptr;
+  {
+    std::lock_guard<std::mutex> lock(lock_last_os_light_);
+
+    auto now_us = ::apollo::cyber::Time::MonoTime().ToMicrosecond();
+    if (last_os_light_ == nullptr ||
+        2000LL * 1000 * 1000 < now_us - ts_last_os_light_) {
+      AWARN << "V2X Traffic Light is too old!";
+      last_os_light_ = nullptr;
+    } else {
+      ADEBUG << "V2X Traffic Light is on time.";
+      last_os_light = std::make_shared<OSLight>();
+      last_os_light->CopyFrom(*last_os_light_);
     }
   }
-  if (signals.size() == 1) {
-    auto single = msg->mutable_single_traffic_light(0);
-    single->set_id(signals[0]->id().id());
-    return true;
-  }
-  auto color = msg->single_traffic_light(0).color();
-  msg->clear_single_traffic_light();
-
-  for (auto i = signals.begin(); i != signals.end(); i++) {
-    auto single = msg->add_single_traffic_light();
-    single->set_id((*i)->id().id());
-    single->set_color(color);
+  // proc planning message
+  bool res_proc_planning_msg = internal_->ProcPlanningMessage(
+      planning_adc.get(), last_os_light.get(), &res_light);
+  if (!res_proc_planning_msg) {
+    return;
   }
-  return true;
+  os_interface_->SendV2xTl4hmiToOs(res_light);
```

Comment:
This function name is really hard to read. 

Once again:
**Names should be descriptive; avoid abbreviation. Always prefer long, human-readable variable/method/class names to short acronyms.**